mysql-test/suite/rpl/t/rpl_show_slave_hosts.test

```diff
@@ -49,4 +49,376 @@ source include/wait_show_condition.inc;
+
+connection master;
+let $status_var= Slaves_connected;
+# Setting the old value for debug_dbug will not start the slave
+let $status_var_value= 1;
+source include/wait_for_status_var.inc;
+show status like "Slaves_connected";
+show status like 'Rpl_semi_sync_master_clients';
+--replace_column 3 'SLAVE_PORT'
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW REPLICA HOSTS;
+
+--echo # Test 9: Stop server_id=3
+--echo # Note:   Reason to simulate the same behaviour is to check that
+--echo #         no replicas get new generated event, so show statement
+--echo #         is expected to be empty.
+--echo # -------------------------------------------
+connection slave2;
+--source include/stop_slave.inc
+--source include/reset_slave.inc
+
+connection master;
+let $status_var= Slaves_connected;
+let $status_var_value= 0;
+source include/wait_for_status_var.inc;
+show status like "Slaves_connected";
+show status like 'Rpl_semi_sync_master_yes_tx';
+show status like 'Rpl_semi_sync_master_clients';
+INSERT INTO t VALUES (5);
+
+--replace_column 3 'SLAVE_PORT'
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW REPLICA HOSTS;
+
+
+--echo # Test 10: Test Sync with rpl_semi_sync_master_timeout=0
+--echo # Note 1:   For asynchronous replication `Gtid_state_sent` is not null,
+--echo #           but `Gtid_state_ack` is null.
+--echo # Note 2:   We want that `rpl_semi_sync_master_timeout=0` be edge case,
+--echo #           by mimicking asynchronous replication, for which
+--echo #           `Gtid_state_ack` should not update (see Note 3).
+--echo # Note 3:   With this option we want `Gtid_state_ack` to reflect
+--echo #           the latest received transaction by the replica[s].
+--echo # -------------------------------------------
+# Test DBUG_SYNC by stopping the slave and testing the new sync point
+# Note we need to stop both slaves in order to control sent ACK for new active transaction
+
+--let $save_default_master_timeout= `SELECT @@GLOBAL.rpl_semi_sync_master_timeout`
+select @@global.rpl_semi_sync_master_timeout;
+show variables like 'rpl_semi_sync_master_enabled';
+show status like 'rpl_semi_sync_master_status';
+show status like 'rpl_semi_sync_master_get_ack';
+set global rpl_semi_sync_master_timeout=0;
+select @@global.rpl_semi_sync_master_timeout;
+# here rpl_semi_sync_master_status=0 means async is taking place (no wait on ACK from slave)
+show variables like 'rpl_semi_sync_master_timeout';
+show status like 'rpl_semi_sync_master_status';
+
+connection slave;
+--let $save_server_2_dbug= `SELECT @@GLOBAL.DEBUG_DBUG`
+SET @@GLOBAL.DEBUG_DBUG="d,synchronize_semisync_slave_reply";
+--source include/start_slave.inc
+SET debug_sync='now WAIT_FOR at_slave_reply';
+
+connection slave2;
+--let $save_server_3_dbug= `SELECT @@GLOBAL.DEBUG_DBUG`
+SET @@GLOBAL.DEBUG_DBUG="d,synchronize_semisync_slave_reply";
+--source include/start_slave.inc
```

Comment:
You'd probably here want
```
SET debug_sync='now WAIT_FOR at_slave_reply';
```
like you have for the `slave` to prove the following claim of both replicas being stopped